hw/ip/aon_timer/doc/_index.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "AON Timer Technical Specification"
+---
+
+# Overview
+
+This document specifies the Always-On ("AON") Timer IP functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+See that document for an overview of how it is integrated into the top level system.
+
+## Features
+
+- Two 32-bit upcounting timers: one timer functions as a wakeup timer, one as a watchdog timer
+- The watchdog timer has two thresholds: bark (generates an interrupt) and bite (resets core)
+
+## Description
+
+### AON Wakeup timer
+
+The always-on wakeup timer operation is straightforward.
+A count starts at 0 and slowly ticks upwards.
+When it reaches / exceeds the wake count, a level wakeup signal is sent to the power manager and a level IRQ is generated to the processor.
+This wakeup signal stays high until it is explicitly acknowledged by software.
+The wakeup timer can be used like a real-time clock for long periods of low-power mode (though it does not give any guarantees of time-accuracy TODO: specify accuracy).
+
+### AON Watchdog timer
+
+The always-on watchdog timer behaves similarly to the wakeup timer.
+It has an independent count starting at 0 which slowly ticks upwards.
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
+Simultaneously, a level IRQ signal is also generated to the processor.
+
+If the system is in a low power state, the wakeup signal asks the power manager to wake the system such that the IRQ can be serviced.
+If the system is not in low power mode, the IRQ is immediately serviced.
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which will reset the whole system.
+This is independent of the IRQ sent as part of the watchdog bark.
+The system reset also resets the always-on timer, so software is not required to directly acknowledge anything after a watchdog reset.
+
+To prevent the watchdog bark or bite, software is expected to periodically reset the count when operating normally.
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has two additional functions not present in the always-on wakeup timer:
+* Watchdog configuration lock
+* Watchdog pause in sleep
+
+Unlike the wakeup timer, the watchdog timer configuration might need to be locked by firmware until the next system reset.
```

Comment:
Could you please specify what registers/functionality is locked by this bit (we still have to pet the watchdog even if the lock bit is set).

Reply:
Just realised that every affected register has `regwen` field in hjson.